x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -22,19 +22,24 @@ import {
   Privilege,
   RuleExecutionStatus,
 } from '../../../../common/detection_engine/schemas/common/schemas';
-import {
+
+import type {
   ElasticsearchClient,
   Logger,
   SavedObjectsClientContract,
 } from '../../../../../../../src/core/server';
-import {
+
+import type {
   AlertInstanceContext,
   AlertInstanceState,
   AlertServices,
-  parseDuration,
 } from '../../../../../alerting/server';
-import { ExceptionListClient, ListClient, ListPluginSetup } from '../../../../../lists/server';
-import {
+
+// We _must_ import from here and not from "alerting/server" above or we will get a memory leak. See https://github.com/elastic/kibana/issues/117255
+import { parseDuration } from '../../../../../alerting/common/parse_duration';
```

Comment:
NOTE: Here is one key area where I had to import as a workaround for the memory leaks found by importing at a higher level within alerting.